### PR TITLE
UFileClose needs a return value. 

### PR DESCRIPTION
**Pull request by gongxun0928**
The close function may have an error and return a return value which less than 0, so UFileClose also needs a return value. The external system checks the return value to determine whether an error has occurred. If an error occurs, the error information is obtained through UFileGetLastError.

fix #ISSUE_Number

---

### Change logs

_Describe your change clearly, including what problem is being solved or what feature is being added._

_If it has some breaking backward or forward compatibility, please clary._

### Why are the changes needed?

_Describe why the changes are necessary._

### Does this PR introduce any user-facing change?

_If yes, please clarify the previous behavior and the change this PR proposes._

### How was this patch tested?

_Please detail how the changes were tested, including manual tests and any relevant unit or integration tests._

### Contributor's Checklist

Here are some reminders and checklists before/when submitting your pull request, please check them:

- [ ] Make sure your Pull Request has a clear title and commit message. You can take [git-commit](https://github.com/cloudberrydb/cloudberrydb/blob/main/.gitmessage) template as a reference.
- [ ] Sign the Contributor License Agreement as prompted for your first-time contribution(*One-time setup*).
- [ ] Learn the [coding contribution guide](https://cloudberrydb.org/contribute/code), including our code conventions, workflow and more.
- [ ] List your communication in the [GitHub Issues](https://github.com/cloudberrydb/cloudberrydb/issues) or [Discussions](https://github.com/orgs/cloudberrydb/discussions) (if has or needed).
- [ ] Document changes.
- [ ] Add tests for the change
- [ ] Pass `make installcheck`
- [ ] Pass `make -C src/test installcheck-cbdb-parallel`
- [ ] Feel free to request `cloudberrydb/dev` team for review and approval when your PR is ready🥳

### CODE REVIEW

**Comment by wenchaozhang-123** [comment]
Need also to change the fileam->close function return type?

**Review by wenchaozhang-123 — APPROVED** [review]

File: src/backend/commands/copyto.c [hunk 1]
```diff
@@ -2167,7 +2167,14 @@ CopyToDirectoryTable(CopyToState cstate)
 						(errcode(ERRCODE_IO_ERROR),
 						 errmsg("Copy directory table to file failed, as file content is not consistent.")));
 
-		UFileClose(file);
+		int ret = UFileClose(file);
```
Comment (jiaqizho) [inline src/backend/commands/copyto.c:2170]: `if (UFileClose(file) < 0)`

`ret` only used in if. and check others?

File: src/backend/commands/copyfrom.c [hunk 2]
```diff
@@ -1146,7 +1146,12 @@ CopyFromDirectoryTable(CopyFromState cstate)
 							(errcode(ERRCODE_IO_ERROR),
 							 errmsg("unable to sync file \"%s\": %s", glob_copystmt->dirfilename, UFileGetLastError(file))));
 
-			UFileClose(file);
+			if (UFileClose(file) < 0)
+				ereport(ERROR,
+							(errcode(ERRCODE_IO_ERROR),
+							 errmsg("failed to close file \"%s\": %s", glob_copystmt->dirfilename, UFileGetLastError(file))));
```
Comment (jiaqizho) [inline src/backend/commands/copyfrom.c:1152]: `failed to close the file`, missing the `the`..

**Review by jiaqizho — APPROVED** [review]
almost LGTM with some comments

File: src/backend/commands/copyto.c (hunk 1 above)
Comment (gongxun0928) [reply src/backend/commands/copyto.c]: done

**Review by jiaqizho — APPROVED** [review]

**Review by wenchaozhang-123 — APPROVED** [review]

**Pull request merged**